onboarding/source/amb_light_service.c

```diff
@@ -15,25 +23,103 @@
+    adc->GxINTCR[group] = fifo_size;
+    /** - Start Conversion */
+    adc->GxSEL[group] = 1 << channel;
+}
+
+void adcGetSingleData(adcBASE_t *adc, unsigned int group, adcData_t *data) {
+    unsigned  buf;
+    adcData_t *ptr = data; 
+
+    /** -  Get conversion data and channel/pin id */
+    buf        = adc->GxBUF[group].BUF0;
+    ptr->value = (unsigned short)(buf & 0xFFFU);
+    ptr->id    = (unsigned short)((buf >> 16U) & 0x1FU); // int to unsigned short
+
+    adc->GxINTFLG[group] = 9U;
+
+    /**   @note The function canInit has to be called before this function can be used.\n
+    *           The user is responsible to initialize the message box.
+    */
+}
+
+uint32_t getLightSensorData(void) {
+	adcData_t adc_data;
+	adcData_t *adc_data_ptr = &adc_data;
+
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversionSelChn(adcREG1, LIGHT_SENSOR, 1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetSingleData(adcREG1, adcGROUP1, adc_data_ptr);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return (adc_data_ptr->value);
+}
+
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if(lightServiceTaskHandle == NULL) {
+        // Create controller task
+        xReturned = xTaskCreate(lightServiceTask,             /* Function that implements the task. */
+                                LIGHT_SERVICE_NAME,            /* Text name for the task. */
+                                LIGHT_SERVICE_STACK_SIZE,      /* Stack size in words, not bytes. */
+                                NULL,                       /* Parameter passed into the task. */
+                                LIGHT_SERVICE_PRIORITY,        /* Priority at which the task is created. */
+                                &lightServiceTaskHandle);     /* Used to pass out the created task's handle. */
+    }
+
+    event_queue = xQueueCreate(EVENT_QUEUE_LEN, sizeof(MEASURE_LIGHT));
 
+    if(xReturned == pdFAIL) {
+        // error: couldnt create light service task
+    }
+    if(event_queue == NULL) {
+        // error: couldnt create event queue
+    }
     /* USER CODE END */
     return 1;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
-
+    light_event_t event;
+    if(xQueueReceive(event_queue, &event, 0) == pdTRUE) {
+        // new event received
+        if(event == MEASURE_LIGHT) {
+            uint32 Light_Sensor_data = getLightSensorData();
+            char ambient_light[100];
+            snprintf(ambient_light, strlen(ambient_light), "%lu\n", Light_Sensor_data);
```

Comment:
The 2nd argument of snprintf is the size of the buffer (100 in this case). Although, since you're only printing the value, your buffer can have a smaller size, like 10.